autotest/test_geospatial_index.py

```diff
@@ -0,0 +1,663 @@
+    """Create a triangular unstructured grid using Delaunay triangulation."""
+    vertices, iverts, xcenters, ycenters = _create_triangular_geometry()
+    ncells = len(iverts)
+    return UnstructuredGrid(
+        vertices=vertices,
+        iverts=iverts,
+        xcenters=xcenters,
+        ycenters=ycenters,
+        top=np.ones(ncells) * 10.0,
+        botm=np.zeros(ncells),
+    )
+
+
+@pytest.fixture
+def simple_vertex_grid():
+    """Create a 10x10 rectangular vertex grid (100 cells, 10x10 each)."""
+    return _create_rectangular_vertex_grid(nrow=10, ncol=10, cell_size=10.0)
+
+
+@pytest.fixture
+def rotated_vertex_grid():
+    """Create a rotated 5x5 rectangular vertex grid."""
+    return _create_rectangular_vertex_grid(nrow=5, ncol=5, cell_size=20.0, angrot=45.0)
+
+
+@pytest.fixture
+def simple_structured_grid():
+    """Create a simple 10x10 structured grid for testing rejection."""
+    nrow, ncol = 10, 10
+    delr = np.ones(ncol) * 10.0
+    delc = np.ones(nrow) * 10.0
+    top = np.ones((nrow, ncol)) * 10.0
+    botm = np.zeros((1, nrow, ncol))
+    return StructuredGrid(delr=delr, delc=delc, top=top, botm=botm)
+
+
+@pytest.fixture
+def layered_unstructured_grid():
+    """Create a 3-layer unstructured grid for 3D testing.
+
+    Grid has 2 cells per layer, 3 layers total = 6 cells.
+    Cell IDs: 0-1 (layer 0), 2-3 (layer 1), 4-5 (layer 2).
+    Z elevations: layer 0 (10-7), layer 1 (7-4), layer 2 (4-1).
+    """
+    vertices, base_iverts, base_xc, base_yc = _create_minimal_geometry()
+
+    nlay = 3
+    ncpl = 2
+
+    # Repeat geometry for each layer
+    iverts = base_iverts * nlay
+    xcenters = list(base_xc) * nlay
+    ycenters = list(base_yc) * nlay
+
+    top = np.array([10.0, 10.0, 7.0, 7.0, 4.0, 4.0])
+    botm = np.array([7.0, 7.0, 4.0, 4.0, 1.0, 1.0])
+
+    return UnstructuredGrid(
+        vertices=vertices,
+        iverts=iverts,
+        xcenters=xcenters,
+        ycenters=ycenters,
+        top=top,
+        botm=botm,
+        ncpl=np.array([ncpl] * nlay),
+    )
+
+
+# ============================================================================
+# Test Index Building
+# ============================================================================
+
+
+def test_rejects_structured_grid(simple_structured_grid):
+    """Test that GeospatialIndex rejects StructuredGrid."""
+    with pytest.raises(ValueError, match="only supports vertex and unstructured"):
+        GeospatialIndex(simple_structured_grid)
+
+
+@pytest.mark.parametrize(
+    "grid_fixture,grid_type,expected_cells,expected_points_per_cell",
+    [
+        ("minimal_vertex_grid", "vertex", 2, 5),
+        ("minimal_unstructured_grid", "unstructured", 2, 5),
+        ("simple_vertex_grid", "vertex", 100, 5),
+    ],
+)
+def test_index_build(
+    grid_fixture, grid_type, expected_cells, expected_points_per_cell, request
+):
+    """Test building index for different grid types."""
+    grid = request.getfixturevalue(grid_fixture)
```

Comment:
nice. so you get named test cases with less repetition than parametrizing via the [object itself and `ids`](https://github.com/modflowpy/flopy/blob/e42f51117d1e0db8d9d7eb22eb8eae713a98262e/autotest/test_grid.py#L1332). TIL.